lorry/component/postgres/officalpostgres/manager.go

```diff
@@ -437,11 +435,16 @@ func (mgr *Manager) checkTimelineAndLsn(ctx context.Context, cluster *dcs.Cluste
 	case localTimeLine == primaryTimeLine:
 		needRewind = false
 	case primaryTimeLine > 1:
-		historys = mgr.getHistory()
+		history = mgr.getHistory(cluster.GetMemberAddr(*cluster.GetLeaderMember()), primaryTimeLine)
 	}
 
-	if len(historys) != 0 {
-		for _, h := range historys {
+	if len(history.History) != 0 {
+		// use a boolean value to check if the loop should exit early
+		flag := false
```

Comment:
rename to `exitFlag` , clearly defines for a variable

Reply:
/done